src/pixelator/pixeldataset.py

```diff
@@ -709,14 +709,10 @@ def graph(
         :raises: KeyError if the provided `component_id` is not found in the edgelist
         """
         if component_id:
-            potential_component = _enforce_edgelist_types(
-                (
-                    self.edgelist_lazy.filter(pl.col("component") == component_id)
-                    .collect()
-                    .to_pandas()
-                )
+            potential_component = self.edgelist_lazy.filter(
```

Comment:
I dropped it since it requires us to materialize the data. It's also not as critical when working on the lazy edge list since, we never bring all of it into memory at the same time (and the reduced memory requirement is the main point of enforcing the types).